src/api/app/models/obs_factory/staging_project.rb

```diff
@@ -207,11 +206,25 @@ def attributes
     end
 
     def build_state
-      return :building if building_repositories.present? || disabled_repositories.present?
+      return :building if building_repositories.present?
       return :failed if broken_packages.present?
       :acceptable
     end
 
+    def checks
+      # for stagings we ignore 'old' checks in disabled repositories
+      repos = disabled_repositories
+      project.checks.to_a.each do |check|
```

Comment:
I don't think you need the `to_a`here. [Project#checks](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/project.rb#L1618) a query collection proxy

Reply:
but what will the return of the each be then?

Reply:
In case of `Status::Check.none` nothing (no iteration happens).
Otherwise the Status::Check objects that were found by the query.

And if we suspect that at some point we might have projects with hundreds of status checks a `find_each` would make sense.

Reply:
the point is I want it to be *one* array that is returned - as I modify the elements. So a find_each is the opposite of what I want.

Reply:
What difference does it make? The iterated elements (`check`) are the same. Only the "container" class changes:

```ruby
irb(main):008:0> report.checks.to_a.first
=> #<Status::Check id: 1, state: "pending", url: "https://foo", short_description: nil, name: "my_check", created_at: "2019-03-01 12:26:36", updated_at: "2019-03-01 12:26:36", status_reports_id: 2>
irb(main):009:0> report.checks.first
=> #<Status::Check id: 1, state: "pending", url: "https://foo", short_description: nil, name: "my_check", created_at: "2019-03-01 12:26:36", updated_at: "2019-03-01 12:26:36", status_reports_id: 2>
irb(main):010:0> report.checks.find_each.first
=> #<Status::Check id: 1, state: "pending", url: "https://foo", short_description: nil, name: "my_check", created_at: "2019-03-01 12:26:36", updated_at: "2019-03-01 12:26:36", status_reports_id: 2>
```

Reply:
because find_each will cast to an array internally if not called with a block - and checks.first is even a SQL query as you access the AREL tree.

But what I want is an array, not a relation and not an enumerator.

Reply:
So to answer my initial question in this thread:

```
Status::Check.limit(10).each { |a| a.state = 'pending' }.class
  Status::Check Load (1.6ms)  SELECT  `status_checks`.* FROM `status_checks` LIMIT 10
=> Array

Reply:
And also 
```
irb(main):001:0> Status::Check.none.each { |a| a.state = 'pending' }.class
=> Array

Reply:
So removed to_a

Reply:
Ah, I didn't realize that you were talking about the return value of the checks method, sorry.